api/http/handler/auth.go

```diff
@@ -78,49 +80,147 @@ func (handler *AuthHandler) handlePostAuth(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	var username = req.Username
-	var password = req.Password
-
-	u, err := handler.UserService.UserByUsername(username)
-	if err == portainer.ErrUserNotFound {
-		httperror.WriteErrorResponse(w, ErrInvalidCredentials, http.StatusBadRequest, handler.Logger)
-		return
-	} else if err != nil {
+	settings, err := handler.SettingsService.Settings()
+	if err != nil {
 		httperror.WriteErrorResponse(w, err, http.StatusInternalServerError, handler.Logger)
 		return
 	}
 
-	settings, err := handler.SettingsService.Settings()
-	if err != nil {
+	u, err := handler.UserService.UserByUsername(req.Username)
+	if err != nil && err != portainer.ErrUserNotFound {
 		httperror.WriteErrorResponse(w, err, http.StatusInternalServerError, handler.Logger)
 		return
 	}
 
-	if settings.AuthenticationMethod == portainer.AuthenticationLDAP && u.ID != 1 {
-		err = handler.LDAPService.AuthenticateUser(username, password, &settings.LDAPSettings)
-		if err != nil {
-			httperror.WriteErrorResponse(w, err, http.StatusInternalServerError, handler.Logger)
+	if (u != nil && u.ID == 1) || settings.AuthenticationMethod == portainer.AuthenticationInternal {
```

Comment:
We recently added the ability to remove the default admin account via https://github.com/portainer/portainer/issues/1921

There is now no "local fallback" option if the admin account is deleted. We believe that we should add some information about this somewhere, I'll add an information message in another commit.

You can remove the `u != nil && u.ID == 1` check though.

Reply:
People should have a last resort for the case when LDAP is down, and I want to change that part to:
IF user exists internally, has admin role and the password is not empty --> authenticate via internal db
ELSE authenticate according to settings.AuthenticationMethod

What you think about this logic?

Reply:
Yes but what if you created an admin account with LDAP enabled and then you remove all the users created previously (the default admin account for example?). You would put yourself in a state where there would be no admin accounts with a password and thus no internal authentification.

cc @ncresswell

Reply:
@AlexJakeGreen I think that we should prevent the removal of the last admin user account with a defined password.